openmls/src/messages/group_info.rs

```diff
@@ -158,21 +158,17 @@ impl GroupInfo {
             return Ok(vec![]);
         }
 
-        // New members compute the interim transcript hash using
-        // the confirmation_tag field of the GroupInfo struct.
-        let confirmed_transcript_hash = self.group_context().confirmed_transcript_hash();
-        let mls_plaintext_commit_auth_data =
-            &InterimTranscriptHashInput::from(self.confirmation_tag());
-        let commit_auth_data_bytes = mls_plaintext_commit_auth_data
-            .tls_serialize_detached()
-            .map_err(LibraryError::missing_bound_check)?;
-
-        crypto
-            .hash(
-                self.group_context().ciphersuite().hash_algorithm(),
-                &[confirmed_transcript_hash, &commit_auth_data_bytes].concat(),
-            )
-            .map_err(LibraryError::unexpected_crypto_error)
+        let interim_transcript_hash = {
+            let input = InterimTranscriptHashInput::from(self.confirmation_tag());
+
+            input.calculate_interim_transcript_hash(
+                crypto,
+                self.group_context().ciphersuite(),
+                self.group_context().confirmed_transcript_hash(),
+            )?
```

Comment:
I adjusted it here but can't find another instance where this would work without making the code weird, i.e., relying on the return value of a assignment to be `()`. In the other cases the code is subtly different.